glotaran/analysis/problem_ungrouped.py

```diff
@@ -37,6 +64,9 @@ def calculate_matrices(
             else:
                 self._calculate_index_independent_matrix(label, dataset_model)
 
+            if dataset_model.global_model:
```

Comment:
this is a bug.

Reply:
`global_model` is a function

Reply:
Yeah, this is due to the nature of how `if` it checks for truthiness.
From the Zen of python (`import this`):
> Explicit is better than implicit.

So using `==` consistently can prevent this bug.
```python
In [1]: def func():
      :     return True
      :

In [2]: if func:
      :     print(True)
      : else:
      :     print(False)
      :
True

In [3]: if func():
      :     print(True)
      : else:
      :     print(False)
      :
True

In [4]: if func == True:
      :     print(True)
      : else:
      :     print(False)
      :
False

In [5]: if func() == True:
      :     print(True)
      : else:
      :     print(False)
      :
True
```

Something else to be aware of is that for historical reasons `bool` is a subclass of `int`:
```python
In [6]: if func() == 1:
      :     print(True)
      : else:
      :     print(False)
      :
True

In [7]: if True == 1:
      :     print(True)
      : else:
      :     print(False)
      :
True
```

So if a function/method can return `bool` or `int` and you need to treat them differently you should use.
```python
In [8]: if isinstance(func(), bool) and func() == True:
       :     print(True)
       : else:
       :     print(False)
       :
True
```

Reply:
I think this issue has been resolved, in all instances it is called correctly now:
![image](https://user-images.githubusercontent.com/3616369/126032328-6c1f9575-03c3-4f25-8f3d-50c54dc933eb.png)
So I'll mark this as resolved.